File: Decorator Pattern/CSharp/Decorator/Soy.cs
using Beverage = Coffee.Beverage;

namespace Decorator 
{
    public class Soy : CondimentDecorator 
    {
        public Soy(Beverage? beverage)
        {
            this.beverage = beverage;
        }

        public override string getDescription() => beverage?.getDescription() + " + Soy";

        public override double cost() => beverage == null ? 0.15 : beverage.cost() + 0.15;
    }
}

File: Decorator Pattern/Cpp/Decorator/Soy.cpp
#pragma once

#include "CondimentDecorator.cpp"

class Soy : public CondimentDecorator {
public:
    Soy(Beverage* beverage) {
        this->beverage = beverage;
    }

    string getDescription() {
        return beverage->getDescription() + " + Soy";
    }

    double cost() {
        return beverage->cost() + 0.15;
    }
};